src/node/node_class.hpp

```diff
@@ -1307,52 +1314,18 @@ Napi::Object ObjectWrap<ClassType>::create_instance_by_schema(Napi::Env env, Nap
         }
 
         schemaObjectConstructor = constructor;
-        Napi::Object constructorPrototype = constructor.Get("prototype").As<Napi::Object>();
 
         // get all properties from the schema
         std::vector<Napi::PropertyDescriptor> properties =
             create_napi_property_descriptors(env, constructorPrototype, schema, false /*redefine*/);
 
-        Napi::Function realmObjectClassConstructor = ObjectWrap<ClassType>::create_constructor(env);
-        bool isInstanceOfRealmObjectClass = constructorPrototype.InstanceOf(realmObjectClassConstructor);
-
-        // Skip if the user defined constructor inherited the RealmObjectClass. All RealmObjectClass members are
-        // available already.
-        if (!isInstanceOfRealmObjectClass) {
-            // setup all RealmObjectClass<T> methods to the prototype of the object
-            for (auto& pair : s_class.methods) {
-                // don't redefine if exists
-                if (!constructorPrototype.HasOwnProperty(pair.first)) {
-                    auto descriptor = Napi::PropertyDescriptor::Function(
-                        env, constructorPrototype, Napi::String::New(env, pair.first) /*name*/, &method_callback,
-                        napi_default | realm::js::PropertyAttributes::DontEnum, (void*)pair.second /*callback*/);
-                    properties.push_back(descriptor);
-                }
-            }
-
-            for (auto& pair : s_class.properties) {
-                // don't redefine if exists
-                if (!constructorPrototype.HasOwnProperty(pair.first)) {
-                    napi_property_attributes napi_attributes =
-                        napi_default |
-                        (realm::js::PropertyAttributes::DontEnum | realm::js::PropertyAttributes::DontDelete);
-                    auto descriptor = Napi::PropertyDescriptor::Accessor<property_getter_callback>(
-                        Napi::String::New(env, pair.first) /*name*/, napi_attributes,
-                        (void*)&pair.second /*callback*/);
-                    properties.push_back(descriptor);
-                }
-            }
-        }
-
         // define the properties on the prototype of the schema object constructor
         if (properties.size() > 0) {
             constructorPrototype.DefineProperties(properties);
         }
 
-        instance = schemaObjectConstructor.New({});
-        if (!instance.InstanceOf(schemaObjectConstructor)) {
-            throw Napi::Error::New(env, "Realm object constructor must not return another value");
-        }
+        instance = ObjectCreate.Call({constructorPrototype}).As<Napi::Object>();
```

Comment:
This change is important. We're creating plain object and setting the prototype instead of calling the constructor provided by the user.